api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -225,15 +262,24 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     dbWorkspace.setCreationTime(now);
     dbWorkspace.setLastModifiedTime(now);
     setCdrVersionId(workspace, dbWorkspace);
-    dbWorkspace = workspaceService.dao.save(dbWorkspace);
+
+    dbWorkspace = workspaceService.getDao().save(dbWorkspace);
+
+    org.pmiops.workbench.db.model.WorkspaceUserRole permissions = new org.pmiops.workbench.db.model.WorkspaceUserRole();
+    permissions.setRole(WorkspaceAccessLevel.fromValue("owner"));
+    permissions.setWorkspace(dbWorkspace);
+    permissions.setUser(user);
+
+    workspaceUserRoleDao.save(permissions);
```

Comment:
I think it would be better to create the WorkspaceUserRole before the save of dbWorkspace, and add it to the collection on the workspace (call the setter or expose an add method.) Then when the workspace gets saved the WorkspaceUserRole entry will be added in the same transaction.